plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/action/GetQueryGroupResponse.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm.action;
+
+import org.opensearch.cluster.metadata.QueryGroup;
+import org.opensearch.core.action.ActionResponse;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Response for the get API for QueryGroup
+ *
+ * @opensearch.experimental
+ */
+public class GetQueryGroupResponse extends ActionResponse implements ToXContent, ToXContentObject {
+    private final List<QueryGroup> queryGroups;
```

Comment:
In what scenarios, Get API might return more than one query groups? While I doubt that is possible, but if it were, it should be BatchGet API?

Reply:
`_wlm/query_group/{name} `returns the single querygroup with {name}, and` _wlm/query_group/ `returns all the querygroups in the system

Reply:
We are probably overloading `get` and `list` together in single API. I am wondering if there is any precedent for that in Opensearch.

Reply:
I came across this: https://opensearch.org/docs/latest/api-reference/index-apis/get-settings/, where `GET /<target-index>/_settings` retrieves all the settings for the specified index and `GET /<target-index>/_settings/<setting>` retrieves a specific setting for the specified index